src/background/utils/db.js

```diff
@@ -379,7 +380,7 @@ function reportBadScripts(ids) {
  * @return {Promise<{ scripts: VMScript[], cache: Object }>}
  */
 export async function getData({ ids, sizes }) {
-  const scripts = ids ? ids.map(getScriptById) : store.scripts;
+  const scripts = ids?.map(getScriptById) ?? [...store.scripts, ...store.removedScripts];
```

Comment:
`[...store.scripts, ...store.removedScripts]` is duplicated in 4 places, so I'd use a function like this:

```js
export function getScriptsByIdsOrAll(ids) {
  return ids
    ? ids.map(id => scriptMap[id])
    : [...dbScripts, ...dbRemovedScripts];
}
```

I can open a new PR with similar code cosmetics after merging this one.